app/src/main/java/com/github/factotum_sdp/factotum/data/SignUpDataSink.kt

```diff
@@ -31,17 +31,33 @@ class SignUpDataSink {
         return authResultFuture.get()
     }
 
-    fun updateUsersList(userUID: String, user: User): Result<String> {
-        val updateUsersResultFuture = CompletableFuture<Result<String>>()
+    fun updateUser(userUID: String, user: User): Result<String> {
+        val updateUserResultFuture = CompletableFuture<Result<String>>()
 
         dbRef.child(DISPATCH_DB_PATH).child(userUID).setValue(user)
             .addOnSuccessListener {
-                updateUsersResultFuture.complete(Result.Success("Success updating users"))
+                updateUserResultFuture.complete(Result.Success(user.name))
             }.addOnFailureListener {
-                updateUsersResultFuture.complete(
+                updateUserResultFuture.complete(
                     Result.Error(IOException("Error updating users", it))
                 )
             }
-        return updateUsersResultFuture.get()
+        return updateUserResultFuture.get()
+    }
+
+    fun fetchClientId(clientId: String): Result<String> {

```

Comment:
As discussed the feature we now want is to verify if the contacts username matches a contact with no account yet, it is not enough to check if client id already exists.